src/UXClient/Components/HierarchyNavigation/HierarchyNavigation.ts

```diff
@@ -118,8 +118,15 @@ class HierarchyNavigation extends Component{
             }).catch(err => this.chartOptions.onError(this.getString("hierarchyNav_error_title"), this.getString("hierarchyNav_hierarchy_load_error"), err instanceof XMLHttpRequest ? err : null));
         }).catch(err => this.chartOptions.onError(this.getString("hierarchyNav_error_title"), this.getString("hierarchyNav_auth_error"), err instanceof XMLHttpRequest ? err : null));
 
+        if (hierarchyNavOptions.selectedHierarchyId) {
+            this.selectedHierarchyName = hierarchyNavOptions.selectedHierarchyId === HierarchySelectionValues.All || hierarchyNavOptions.selectedHierarchyId === HierarchySelectionValues.Unparented ? 
+                                hierarchyNavOptions.selectedHierarchyId
+                                : Object.values(this.envHierarchies).find(h => h["id"] === hierarchyNavOptions.selectedHierarchyId)["name"];
```

Comment:
what if the hierarchy name or id is not present.  like what if find returns null?

Reply:
if the selectedHierarchyId is not present the default one (All hierarchies) preserved. However, you made a good point that if find returns null (e.g. if the hierarchy no longer exists), I should still preserve the default one (All hierarchies)